semgrep-core/src/core/ast/AST_generic_helpers.ml

```diff
@@ -128,6 +128,26 @@ let name_of_ids xs =
           name_info = empty_id_info ();
         }
 
+let add_suffix_to_name suffix name =
+  match name with
+  | Id (first_id, _) -> name_of_ids [ first_id; suffix ]
+  | IdQualified ({ name_last; name_middle; _ } as q_info) ->
+      let new_name_middle =
+        let middle_quals =
+          match name_middle with
+          | Some (QDots qualifiers) -> qualifiers
+          | Some (QExpr _) -> raise Impossible
```

Comment:
If this is a more generic routine we may want to handle this case, in the event that someone working on Ruby, etc...